contracts/interfaces/draft-IERC6909.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC165} from "../utils/introspection/IERC165.sol";
+
+/**
+ * @dev Required interface of an ERC-6909 compliant contract, as defined in the
+ * https://eips.ethereum.org/EIPS/eip-6909[ERC].
+ */
+interface IERC6909 is IERC165 {
+    /**
+     * @dev Emitted when the allowance of a `spender` for an `owner` is set by a call to `approve` for a
+     * token of type `id`. The new allowance is `amount`.
```

Comment:
> Note that the ERC states:
> 
> > MUST be logged when the allowance is set by an owner.
> 
> So it may not be strictly done through `approve`

Meaning there may be a non ERC function used to update the allowance by the owner?

Reply:
Correct. A developer may extend the standard and use another function to set the allowance. For example, something like `approveWithSignature`

Reply:
Ok good point. Will generalize it for that consideration.